torch/backends/cudnn/rnn.py

```diff
@@ -190,7 +196,11 @@ def forward(fn, input, hx, weight, output, hy):
         handle = cudnn.get_handle()
         fn.datatype = cudnn._typemap[input.type()]
         is_input_packed = fn.batch_sizes is not None
-
+        if is_input_packed:
+            warnings.warn(
+                "Warning: persistent algorithm not supported for variable length input."
+                "Switching to standard")
+            fn.persistent = False  # persistent algo is not supported for variable length input
```

Comment:
Changes in sequence length are usually well behaved (it's the number of gemms called in standard algo, or the loop count in persistent), but changes in batch size/hidden size can alter performance a lot. Cublas releases are about every 6 months, give or take, but cudnn releases are independent and can be built with some intermediate versions of cublas. 
I can put in benchmarking persistent and standard algos if cudnn.benchmark is set, and we will seek input of power users like @jekbradbury to see if it's enough or if they want some better approach.